### DISCUSSION TITLE
Support for Basic Authentication in Next.js and/or Vercel hosting

### DISCUSSION THREAD

**Discussion by OZZlE** [vercel/next.js#17719] [Ideas]
# Feature request

## Is your feature request related to a problem? Please describe.

No support for Basic Authentication via node server. Tried https://github.com/labd/nextjs-basic-auth-middleware/issues/6 but it seems to require no pages are SSR and then using nextjs is pointless...

Cannot have staging websites free on google..

## Describe the solution you'd like

Support for Basic Authentication preferable where you can turn it on/off based on environment/hostname and perhaps a list of credentials. Same as Nginx or Apache supports.

## Describe alternatives you've considered

Nextjs via Express.js (don't know if it works): https://dev.to/ut0n69/how-to-add-basic-auth-in-next-js-apps-with-firebase-hosting-and-ssr-4d8a <- seems firebase hosted specific

**Comment by OZZlE** [comment]
I managed to solve it through hosting NextJS via Express.js https://stackoverflow.com/questions/64316886/how-do-i-add-basic-authentication-to-nextjs-node-server

But I think it would be good if it could be done purely with NextJS?

**Comment by afzalsayed96** [comment]
![Screen Shot 2020-11-09 at 4 33 07 PM](https://user-images.githubusercontent.com/14029371/98534740-ea8fa800-22aa-11eb-9e4f-a90eb1fe63c3.png)
Using custom server for basic authentication seems like quite huge trade-off.

Hoping Next.js gets built-in support for basic auth

**Comment by OZZlE** [comment]
Ignoring Next.js itself is it offered by Vercel hosting (of Next.js app) somehow?

**Reply by f3dd3rn** [reply]
Does [this](https://vercel.com/blog/protecting-deployments) answer your question @OZZlE?

**Reply by OZZlE** [reply]
@f3dd3rn yes great, thanks!
